### PR TITLE
Separate Object JSON Data

### LINKED ISSUE #2: Duplicate transitions
As [reported here](https://onehouronelife.com/forums/viewtopic.php?pid=2653#p2653), some objects have duplicate transitions when combining the same object together. For example thread or reeds.

### LINKED ISSUE #7: Need to interpret categories
I haven't parsed out categories yet. This means for example that you can't see the uses for https://kazetsukai.github.io/onetech/#210/Full-Water-Pouch on various things that need water. Possibly even more crucial is seeing kindling sources, i.e. https://kazetsukai.github.io/onetech/#132/Yew-Branch should show that you can turn it into kindling because it should be a https://kazetsukai.github.io/onetech/#367/%40-Kindling-Source

### LINKED ISSUE #9: Fix Search Autocomplete Sorting
If you search for "Carrot", the actual object is not easily found in the list. The autocompletion list should be sorted based on the length of the name so "Carrot" would show up first.

### PR DESCRIPTION
**Pull request by ryanb**
## Separate JSON Files

I've refactored the `process.js` script into separate classes and changed the JSON output. It now creates a small `objects.json` file and a separate `objects/{id}.json` file for each object which is requested when going to that object.

This has several benefits:
* Smaller initial objects file (from 1.3MB to 35KB)
  * We can get this even smaller by using nested arrays instead of objects
* Less client-side processing
  * Transitions are included in each object
  * Additional info like the tech tree can be included as well

## Object Complexity

The process script also calculates the complexity of each object. This represents the amount of work involved in crafting a given object. Natural objects have a complexity of 1. Combining objects together adds their complexity + 1 (for example a sharp stone is complexity 3). A Dry Deep Well is the most complex at 556.

The complexity calculation doesn't count tools more than once. For example, if you used a flint chip to skin a rabbit, it won't count it again when cutting the fur into pieces.

I'm using this complexity in a couple ways:
* Objects on the index page are sorted by complexity (descending)
  * This has the nice benefit of newer (more complex) objects will appear near the top
  * I still think this sort order could be improved
* Transitions are sorted by their complexity (ascending)
  * This means the easiest way to reach the item will be the first one in the list
* I will use this complexity in the upcoming tech tree branch

## Other Changes

* The search sort order is based on name length which closes #9
* Duplicate transitions no longer show up. Closes #2 
* Transitions are added to objects within the category. Closes #7

**Update**: I rewrote this PR since so much changed over time. I think this is complete now.

### CODE REVIEW

File: process/src/GameData.js [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+  }
+
+  staticDir(dir) {
+    const path = "../static/" + dir;
+    if (!fs.existsSync(path)) fs.mkdirSync(path);
+  }
+
+  saveJSON(path, data) {
+    const minPath = "../static/" + path;
+    const prettyPath = "../static/pretty-json/" + path;
```
Comment (Kazetsukai) [inline process/src/GameData.js]: For any future work, pretty-json doesn't need to be maintained, I had it in there as a little hack to see the diffs a little nicer when committing (unless of course you are using it later on in here that I haven't seen yet)

File: src/components/ObjectSearch.vue [hunk 2]
```diff
@@ -1,6 +1,6 @@
 <template>
   <div class="objectSearch">
-    <VueSelect label="name" :options="objects" v-model="tmpSelect" :on-change="selectObject">
+    <VueSelect label="name" :options="sortedObjects" v-model="selectedObject" :on-change="selectObject">
```
Comment (Kazetsukai) [inline src/components/ObjectSearch.vue]: Just trying to remember why I didn't bind the model straight to the prop. I think I was getting an error but can't remember, so I'll do a quick smoke test around this before merging.

File: process/src/GameData.js (hunk 1 above)
Comment (ryanb) [reply process/src/GameData.js]: I am generating pretty JSON for convenience in debugging. They are in .gitignore so not checked in.

File: src/components/ObjectSearch.vue (hunk 2 above)
Comment (Kazetsukai) [reply src/components/ObjectSearch.vue]: Looks good as far as I can see, all the names are updating.

**Pull request merged**